### DISCUSSION TITLE
Optimization of gifs

### DISCUSSION THREAD

**Discussion by AquaCylinder** [ImageMagick/ImageMagick#2693] [Help] [answered]
Hi!
So I have been trying to optimize my gifs using IM as I haven't found any other program I can acces this easily through batch files. I have been trying to optimize but I'm not getting anywhere maybe I'm using the program wrong.
My question is what optimizations has the most gain to be had in the overal file size? of for example this gif:
![Breathing+MovingCrystal-Blue-256x256](https://user-images.githubusercontent.com/56627792/95378867-d55ade80-08e4-11eb-98f2-bb13cff0ba85.gif)
I have tried these components get it down a little [ `-layers OptimizeTransparency -layers optimize +remap` ] but I'm pretty sure this isn't the way to use them. These are the lines of code I used to test in powershell:
`cd (Get-Clipboard).Replace('"',"")`
`	magick gif.gif -layers OptimizeTransparency -layers optimize +remap output-gif.gif`
Last time you really helped me and I want to again thank you for that and thank you in advance!

Edit: I use Windows 10 and IM 7.0.10-19Q16 x64

**Comment by fmw42** [comment]
I do not understand your +remap. I do not think it does anything. You should create color table image of 256 colors or less and remap each frame. The fewer the colors, the smaller the file size.

`magick gif.gif -coalesce -remap colortable_image -layers OptimizeTransparency -layers optimize output-gif.gif`

**Reply by AquaCylinder** [reply]
Thx, i didn't know why that was there either XD so how would i make the colortable_image? And would you know why the `-layers optimizetransparency` doesn't have much impact i think i saw a example that had a almost 40% reduction?
Also i found a website that has a very good compressor [https://compress-or-die.com/gif](url) and i wanted to get kinda close to that reduction, it has a 60% reduction with minimal effect on the looks at least none that are very noticable. 
![Breathing+MovingCrystal-Blue-256x256](https://user-images.githubusercontent.com/56627792/95382275-b1e66280-08e9-11eb-9610-b32a98f83213.gif)
So the website has something called lossy, would you have any idea what would be a equivalant to this in IM?
I know i'm asking much as you abviously don't know what's behind the scenes on this website, i thought like why not try?

**Reply by fmw42** [reply]
YOur link does not work for me -- 404 Error. I know nothing about lossy gif compression. I have never heard of that for GIF.

You can make a color table using 

magick image -depth 8 -unique-colors colorable.gif

or

magick image -depth 8 +dither -colors XXX -unique-colors colorable.gif

where XXX is the number of desired colors.

**Reply by AquaCylinder** [reply]
Thank you, and sorry about the link.
Thank you for the explenation! So on the documenton i found a little program called Giflossy which is perfect for what i want, it uses lossy compression to get the size down to more than half! Sadly it isn't windows supported BUT it's based on gifsicle which does support windows 10. So i now use `gifsicle -O3 --colors 256 --lossy=80 -o output.gif input.gif` to get the size down. I am pretty sure that the website that i was talking about is using gifsicle as this has the same inputs and get's the size down about the same!

**Comment by AquaCylinder** [comment, marked as answer]
So on the documenton i found a little program called Giflossy which is perfect for what i want, it uses lossy compression to get the size down to more than half! Sadly it isn't windows supported BUT it's based on gifsicle which does support windows 10. So i now use `gifsicle -O3 --colors 256 --lossy=80 -o output.gif input.gif` to get the size down. I am pretty sure that the website that i was talking about is using gifsicle as this has the same inputs and get's the size down about the same!

**Reply by Fred-Vatin** [reply]
This site https://ezgif.com/optimize/ use Gifsicle and Lossy GIF. It would be nice to get the same result with IM.

**Reply by AquaCylinder** [reply]
Well If it does then you can download gifsicle and access it through IM. It has worked perfectly for me! But to have it standard in IM would be nice too. tho I would have to change up my code :P
